### ISSUE TITLE
Code marked as unreachable in edge case with type hint and loop

### ISSUE THREAD

**Issue by krooq** [microsoft/pylance-release#334]
## Environment data
-   VSCode version: 1.48.2
-   Language Server version: 2020.9.0 (3 September 2020) (pyright 60eb2cca)
-   OS and version:  Windows_NT x64 10.0.19042
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.4 (miniconda)

## Expected behaviour

Code that is reachable is marked as such.

## Actual behaviour
In this odd edge case.  
Type hinted code that is called in a loop is marked as unreachable when it otherwise would not be marked as such.

```python
class Foo:
    def foo():
        raise NotImplementedError

def bad():
    f: Foo = Foo() 
    for _ in range(1):
        f.foo()
        print("Code is unreachable") 

def ok():
    f = Foo() 
    for _ in range(1):
        f.foo()
        print("Reachable") 

def also_ok():
    f: Foo = Foo() 
    f.foo()
    print("Reachable") 
```

**Comment by erictraut** [comment]
Thanks for reporting this. This has already been addressed in Pyright (the type checker that Pylance is based on), and it will be fixed in the next release of Pylance as well.

As a workaround, if you provide an explicit return type annotation for `foo`, Pylance will do the right thing.

**Comment by huguesv** [comment]
This issue has been fixed in version 2020.9.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202094-10-september-2020
